h2o-core/src/main/java/water/parser/FVecParseWriter.java

```diff
@@ -83,6 +85,10 @@ public FVecParseWriter(Vec.VectorGroup vg, int cidx, Categorical[] categoricals,
     _cidx = cidx;
     _vg = vg;
     _chunkSize = chunkSize;
+    _monitor = monitor;
+    if (_monitor != null) {
+      _monitor.register(this);
```

Comment:
This will increase memory usage... IMHO by a lot: essentially this causes the `_nvs` data to stay in memory even when we are not working with that chunk anymore.

Reply:
_nvs is set to null after chunks are closed so the data should be freed from memory once not needed, this might however keep the FVecParseWriter in memory for longer
any suggestion of a better way of how to keep track of all the created chunks and their size is welcome

Reply:
yep - I just checked that, close will kill the _nvs array, mea culpa

in that case - do we need to to expose `getNvs`, can there eg. be a race condition in https://github.com/h2oai/h2o-3/pull/3980/files#diff-ad31e0d8480e348d69e75a53b763a9ffR140? 

you check that getNvs is not null, but that can change before the next method is executed - synchronizing on writers won't help

Reply:
you are correct, there is a tricky race condition, will look into it

Reply:
fixed